File: webconsole/src/main/java/com/bytex/snamp/web/serviceModel/charts/ScaleOutChart.java
package com.bytex.snamp.web.serviceModel.charts;

import com.bytex.snamp.supervision.elasticity.ScalingMetrics;
import org.codehaus.jackson.annotate.JsonTypeName;

/**
 * @author Roman Sakno
 * @version 2.1
 * @since 2.0
 */
@JsonTypeName("scaleOut")
public final class ScaleOutChart extends ScalingRateChart {
    public ScaleOutChart() {
        super(ScalingMetrics::scaleOut);
    }

    @Override
    protected NumericAxis createAxisY() {
        final NumericAxis axis = new NumericAxis();
        axis.setUOM("operations");
        axis.setName("upscale");
        return axis;
    }
}

File: webconsole/src/main/resources/_webapp/src/app/charts/model/charts/scale.out.chart.ts
import { ScalingRateChart } from "../scaling.rate.chart";
import { NumericAxis } from "../axis/numeric.axis";
import { AbstractChart } from "../abstract.chart";

export class ScaleOutChart extends ScalingRateChart {
    get type():string {
        return AbstractChart.SCALE_OUT;
    }

    public createDefaultAxisY() {
        let _axis:NumericAxis = super.createDefaultAxisY();
        _axis.unitOfMeasurement = "operations";
        _axis.name = "upscale";
        return _axis;
    }
}
